packages/paste-core/components/disclosure/src/DisclosureHeading.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import {Heading, HeadingProps, HeadingPropTypes} from '@twilio-paste/heading';
+import {Box} from '@twilio-paste/box';
+import {ChevronDownIcon, ChevronDownIconProps} from '@twilio-paste/icons/esm/ChevronDownIcon';
+import {DisclosurePrimitive, DisclosurePrimitiveProps} from '@twilio-paste/disclosure-primitive';
+import {DisclosureContext, Variants} from './Disclosure';
+
+export interface DisclosureHeadingProps extends Omit<DisclosurePrimitiveProps, 'baseId' | 'toggle'> {
+  children: NonNullable<React.ReactNode>;
+  as: HeadingProps['as'];
+  marginBottom?: HeadingProps['marginBottom'];
+  variant: HeadingProps['variant'];
+}
+
+interface StyledDisclosureHeadingProps extends Omit<DisclosureHeadingProps, 'as'> {
+  renderAs: HeadingProps['as'];
+  customDisabled?: boolean;
+  customFocusable?: boolean;
+  disclosureVariant: Variants;
+}
+
+const getIconSize = (variant: HeadingProps['variant']): ChevronDownIconProps['size'] => {
+  switch (variant) {
+    case 'heading10':
+      return 'sizeIcon90';
+    case 'heading20':
+      return 'sizeIcon70';
+    case 'heading30':
+      return 'sizeIcon60';
+    case 'heading40':
+      return 'sizeIcon40';
+    case 'heading50':
+      return 'sizeIcon30';
+    case 'heading60':
+    default:
+      return 'sizeIcon20';
+  }
+};
+
+const getVariantStyles = (variant: Variants): {} => {
+  switch (variant) {
+    case 'contained':
+      return {
+        paddingBottom: 'space50',
+        paddingLeft: 'space40',
+        paddingRight: 'space40',
+        paddingTop: 'space50',
+      };
+    default:
+      return {};
+  }
+};
+
+const StyledDisclosureHeading = React.forwardRef<HTMLDivElement, StyledDisclosureHeadingProps>(
+  ({children, marginBottom, renderAs, disclosureVariant, customDisabled, customFocusable, variant, ...props}, ref) => {
+    return (
+      <Heading
+        as={renderAs}
+        marginBottom={disclosureVariant === 'contained' ? 'space0' : marginBottom}
```

Comment:
I noticed on the examples, the default `marginBottom` makes it look like theres a gap under the heading:
![image](https://user-images.githubusercontent.com/1350081/86033917-f747e980-b9f6-11ea-91dc-99bd63037c19.png)

Should we just set the margin to `space0`, and have some padding in the `DisclosureContent`?

Reply:
I went back and forth on this one, and ended up here.

The reasoning I’m making is that if you’re wanting to just make sections collapsed but keep the typography styles the same as other pages, why _wouldn’t_ you want the same margins the heading gives you? Given that these aren’t generally on their own, there’s usually content around them and keep the vertical rhythm consistent across pages should be desirable. 

The thing about Heading is that we intentionally designed a certain sized gap below each variant because that’s what we thought would look best for these headings when they’re used as normal typographical elements

Reply:
That checks out. You would want the same margins that heading gives you. Keeping that vertical rhythm consistency is important.